app/controllers/spree/shipwire_webhook_controller.rb

```diff
@@ -0,0 +1,34 @@
+class Spree::ShipwireWebhookController < ActionController::Base
+  respond_to :json
+
+  before_action :validate_key, except: :subscribe
```

Comment:
I think this looks good. It's a basic check it shouldn't hurt other code and it's covered with specs.